### PR TITLE
MM-25669 - Matterbuild cutplugin always requires --force for tags that match an existing tag prefix

### PR DESCRIPTION
**Pull request by alifarooq0**
#### Summary
Iterate over the returned github refs to check if one of them matches the exact tag.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-25669

### CODE REVIEW

**Review by jaydeland — COMMENTED** [review]
LGTM

**Comment by alifarooq0** [comment]
just an FYI for @hanzei that we're making this change (not requesting for a review :))

File: server/plugin_release.go [hunk 1]
```diff
@@ -127,16 +127,20 @@ func getReleaseByTag(ctx context.Context, client *GithubClient, owner, repositor
 // createTag creates a new tag at the given commit for the repository.
 // Returns ErrTagExists if tag already exists, nil if successful and an error otherwise.
 func createTag(ctx context.Context, client *GithubClient, owner, repository, tag, commitSHA string) error {
-	refs, _, err := client.Git.GetRefs(ctx, owner, repository, fmt.Sprintf("tags/%s", tag))
+	tagRef := fmt.Sprintf("tags/%s", tag)
+	refs, _, err := client.Git.GetRefs(ctx, owner, repository, tagRef)
```
Comment (metanerd) [inline server/plugin_release.go:131]: go-github will throw an error and a response when there is no ref found. I ran into this inconsistency with the rest of the library already a few times. https://github.com/mattermost/mattermost-mattermod/blob/201158c1d4414b699563a30945a851744bfc361e/server/github.go#L215

**Review by metanerd — CHANGES_REQUESTED** [review]
LGTM otherwise! GJ!

File: server/plugin_release.go (hunk 1 above)
Comment (alifarooq0) [reply server/plugin_release.go:131]: @metanerd yep that's correct, which is why i also check for the `err` here and check for the `StatusNotFound` status code, as I want to create the tag if it doesn't exist.

Also even if `refs` is `nil` it's still iterable and the loop below should still work fine.

Was that the concern here?

Comment (metanerd) [reply server/plugin_release.go:131]: Oh my, yes! Nicely done!
guy

**Review by metanerd — APPROVED** [review]

**Pull request merged**